main/src/ca/uwaterloo/flix/api/lsp/provider/CodeActionProvider.scala

```diff
@@ -218,6 +222,44 @@ object CodeActionProvider {
     command = None
   )
 
+  /**
+    * Returns a code action that proposes to import corresponding Java class.
+    *
+    * For example, if we have:
+    *
+    * {{{
+    *   def foo(): = new File("data.txt")
+    * }}}
+    *
+    * where the undefined class `File` is a valid Java class, this code action proposes to add:
+    * {{{
+    *   import java.io.File
+    * }}}
+    */
+  private def mkImportJava(name: String, uri: String, em: EnclosingMod): List[CodeAction] = {
```

Comment:
I renamed EnclosingMod to AnchorPosition and changed its definition, so you will need to merge/rebase and then refactor.